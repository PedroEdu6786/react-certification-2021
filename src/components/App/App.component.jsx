import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import HomePage from '../../pages/Home';
import VideoDetails from '../../pages/VideoDetails';
import Layout from '../Layout';
import ThemeContentProvider from '../../providers/ThemeContentProvider';
import FavoriteVideos from '../../pages/FavoriteVideos';
import PrivateRoute from '../PrivateRoute/PrivateRoute.component';

function App() {
  return (
    <BrowserRouter>
      <ThemeContentProvider>
        <Layout>
          <Switch>
            <PrivateRoute path="/favorites">
              <FavoriteVideos />
            </PrivateRoute>
            <Route path="/:videoId">
              <VideoDetails />
            </Route>
            <Route exact path="/">
              <HomePage />
            </Route>
          </Switch>
        </Layout>
      </ThemeContentProvider>
    </BrowserRouter>
  );
}

export default App;
