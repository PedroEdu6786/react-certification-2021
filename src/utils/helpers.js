import axios from 'axios';
import { YOUTUBE_MAX_RESULTS } from './constants';

const SNIPPET = 'snippet';
const VIDEO = 'video';

// youtube api axios config
export const youtubeClient = axios.create({
  baseURL: 'https://www.googleapis.com/youtube/v3',
  params: {
    part: SNIPPET,
    key: process.env.REACT_APP_YOUTUBE_API_KEY,
    maxResults: YOUTUBE_MAX_RESULTS,
    type: VIDEO,
  },
});

// build query params helper function
/*
  example:
  const queryParams = buildQueryParams({
    q: query,
  });
*/
export const buildQueryParams = (data) => {
  return Object.entries(data)
    .map(([key, value]) => {
      let finalValue = value;
      if (Array.isArray(value)) {
        finalValue = value.join(',');
      }

      return `${key}=${encodeURI(finalValue)}`;
    })
    .join('&');
};
