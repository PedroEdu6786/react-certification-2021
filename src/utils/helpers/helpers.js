/*
  build query params helper function
  example:
  const queryParams = buildQueryParams({
    q: query,
  });
*/
export const buildQueryParams = (data) => {
  return Object.entries(data)
    .map(([key, value]) => {
      let finalValue = value;
      if (Array.isArray(value)) {
        finalValue = value.join(',');
      }

      return `${key}=${encodeURI(finalValue)}`;
    })
    .join('&');
};
