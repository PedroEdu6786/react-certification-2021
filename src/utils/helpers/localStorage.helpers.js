/* 
  gets item from localStorage
  params:
    itemKey: key used to store data in localStorage

  return: item found on local storage || if not found returns undefined
*/
export const getFromLocalStorage = (itemKey) => {
  let data = localStorage.getItem(itemKey) || undefined;
  if (!data) return data;
  data = JSON.parse(data);

  return data;
};

/* 
  adds item to localStorage
  params:
    data: item to store in localStorage
    itemKey: key used to store data in localStorage

  return: void
*/
export const addToLocalStorage = (data, itemKey) => {
  localStorage.setItem(itemKey, JSON.stringify(data));
};
