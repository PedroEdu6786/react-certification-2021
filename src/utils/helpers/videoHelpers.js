import {
  removeFavoriteVideosAction,
  setFavoriteVideosAction,
} from '../../providers/VideoProvider/VideoProvider.actions';

const addVideo = (videoData) => {
  let favoriteList = localStorage.getItem('REACT-CHALLENGE-FAVORITE-VIDEOS') || [];
  favoriteList = favoriteList.length > 0 ? JSON.parse(favoriteList) : [];
  favoriteList.push(videoData);
  localStorage.setItem('REACT-CHALLENGE-FAVORITE-VIDEOS', JSON.stringify(favoriteList));

  return favoriteList;
};

const removeVideo = (videoId) => {
  let favoriteList = localStorage.getItem('REACT-CHALLENGE-FAVORITE-VIDEOS') || [];

  let updatedList = [];
  if (favoriteList.length > 0) {
    favoriteList = JSON.parse(favoriteList);
    updatedList = favoriteList.filter((video) => video.id.videoId !== videoId);
  }

  localStorage.setItem('REACT-CHALLENGE-FAVORITE-VIDEOS', JSON.stringify(updatedList));

  return updatedList;
};

export const addFavoriteVideo = (videoData, globalDispatch) => {
  const data = addVideo(videoData);
  globalDispatch(setFavoriteVideosAction(data));
};

export const removeFavoriteVideo = (videoId, globalDispatch) => {
  const data = removeVideo(videoId);
  globalDispatch(removeFavoriteVideosAction(data));
};

export const findVideoById = (videos, videoId) => {
  return videos.find((video) => video.id.videoId === videoId);
};
