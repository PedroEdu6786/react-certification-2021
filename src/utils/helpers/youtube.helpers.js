import axios from 'axios';
import { YOUTUBE_MAX_RESULTS } from '../constants';

const SNIPPET = 'snippet';
const VIDEO = 'video';

// youtube api axios config
export const youtubeClient = axios.create({
  baseURL: 'https://www.googleapis.com/youtube/v3',
  params: {
    part: SNIPPET,
    key: process.env.REACT_APP_YOUTUBE_API_KEY,
    maxResults: YOUTUBE_MAX_RESULTS,
    type: VIDEO,
  },
});
