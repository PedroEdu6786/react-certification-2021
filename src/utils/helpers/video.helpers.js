import { setFavoriteVideosAction } from '../../providers/VideoProvider/VideoProvider.actions';
import { REACT_CHALLENGE_FAVORITE_VIDEOS } from '../constants';
import { addToLocalStorage, getFromLocalStorage } from './localStorage.helpers';

/* 
  adds a new favorite video to localStorage
  params:
    videoData: videoObject
    globalDispatch: VideoProvider dispatcher

  return: void
*/
export const addFavoriteVideo = (videoData, globalDispatch) => {
  const videoList = getFromLocalStorage(REACT_CHALLENGE_FAVORITE_VIDEOS) || [];
  videoList.push(videoData);
  addToLocalStorage(videoList, REACT_CHALLENGE_FAVORITE_VIDEOS);
  globalDispatch(setFavoriteVideosAction(videoList));
};

/* 
  removes a favorite video from localStorage
  params:
    videoId: videoObject id
    globalDispatch: VideoProvider dispatcher
  return: void
*/
export const removeFavoriteVideo = (videoId, globalDispatch) => {
  const data = getFromLocalStorage(REACT_CHALLENGE_FAVORITE_VIDEOS);
  const updatedList = data.filter((video) => video.id.videoId !== videoId);
  addToLocalStorage(updatedList, REACT_CHALLENGE_FAVORITE_VIDEOS);

  globalDispatch(setFavoriteVideosAction(updatedList));
};

/* 
  finds video by id
  params:
    videos: array of videos object
    videoId: videoId of video to find
  return: void
*/
export const findVideoById = (videos, videoId) => {
  return videos.find((video) => video.id.videoId === videoId);
};
