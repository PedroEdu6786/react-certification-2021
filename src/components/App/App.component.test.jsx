import React from 'react';
import { act } from 'react-dom/test-utils';
import { findAllByRole, fireEvent, render, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import App from './App.component';
import { youtubeClient } from '../../utils/helpers/youtube.helpers';
import videos from '../../mocks/videos.json';

const build = () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/']}>
      <App />
    </MemoryRouter>
  );

  return {
    container,

    /** Queries for multiple elements */
    videoCards: () => findAllByRole(container, 'article'),
  };
};

describe('App component test', () => {
  it('renders', async () => {
    await act(async () => {
      await build();
    });
  });

  it('navigates to video details screen', async () => {
    youtubeClient.mockImplementationOnce(() => Promise.resolve({ data: videos }));
    let videoCards = null;
    let container = null;
    await act(async () => {
      ({ videoCards, container } = await build());
    });

    let data = null;
    await waitFor(async () => {
      data = await videoCards();
    });

    const videoCard = data[1].children[0];

    act(() => {
      fireEvent.click(videoCard);
    });

    const expectedPath = `/${videos.items[1].id.videoId}`;

    expect(window.location.pathname).toEqual(expectedPath);
    expect(container).toMatchSnapshot();
  });
});
