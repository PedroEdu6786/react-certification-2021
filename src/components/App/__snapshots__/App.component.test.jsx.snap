// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test navigates to video details screen 1`] = `
<div>
  <main
    class="sc-bdnxRM flyXsy"
  >
    <header
      class="sc-bdnxRM sc-jSFjdj sc-crzoAE flyXsy fkwVUy ZaLre"
      data-testid="header"
      direction="column"
    >
      <div
        class="sc-bdnxRM sc-jSFjdj sc-dIsUp flyXsy fkwVUy fMXYSw"
        direction="column"
      >
        <nav
          class="sc-bdnxRM sc-jSFjdj sc-ksluID flyXsy fkwVUy eOCFBL"
          direction="column"
        >
          <a
            class="sc-iemWCZ gtMcKg"
            href="/"
          >
            Home
          </a>
        </nav>
        <div
          class="sc-bdnxRM sc-hBMUJo flyXsy fjglmB"
          data-testid="overlay"
        />
        <div
          class="sc-bdnxRM sc-jSFjdj sc-fnVZcZ flyXsy fkwVUy jxnxQS"
          direction="column"
        >
          <button
            class="sc-bdnxRM sc-bqGGPW flyXsy ebSmms"
          >
            <svg
              class="sc-dIvrsQ kZNHRm"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
              />
            </svg>
          </button>
          <form
            class="sc-bdnxRM sc-jSFjdj sc-dlMDgC flyXsy fkwVUy iwpgqZ"
            direction="column"
          >
            <div
              class="sc-bdnxRM fzigNa"
            >
              <svg
                color="#BDBDBD"
                data-testid="input-search"
                fill="currentColor"
                height="1.125rem"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(189, 189, 189);"
                viewBox="0 0 16 16"
                width="1.125rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.7 13.3l-3.81-3.83A5.93 5.93 0 0 0 13 6c0-3.31-2.69-6-6-6S1 2.69 1 6s2.69 6 6 6c1.3 0 2.48-.41 3.47-1.11l3.83 3.81c.19.2.45.3.7.3.25 0 .52-.09.7-.3a.996.996 0 0 0 0-1.41v.01zM7 10.7c-2.59 0-4.7-2.11-4.7-4.7 0-2.59 2.11-4.7 4.7-4.7 2.59 0 4.7 2.11 4.7 4.7 0 2.59-2.11 4.7-4.7 4.7z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <input
              class="sc-dlnjwi sc-hHEiqL inzMHh dVsAGq"
              value="wizeline"
            />
          </form>
        </div>
        <div
          class="sc-bdnxRM sc-jSFjdj sc-fFSPTT flyXsy fkwVUy cZbFsj"
          direction="column"
        >
          <button
            class="sc-gtsrHT sc-bkbkJK kViayw"
            data-testid="themeButton"
          >
            <svg
              color="#BDBDBD"
              data-testid="lightTheme"
              fill="none"
              height="2.5rem"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              style="color: rgb(189, 189, 189);"
              viewBox="0 0 24 24"
              width="2.5rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"
              />
            </svg>
          </button>
          <button
            class="sc-gtsrHT sc-kEqXSa kViayw iHHQOg"
          >
            Login
          </button>
        </div>
      </div>
    </header>
    <div
      class="sc-bdnxRM sc-gKAaRy sc-pNWdM sc-jrsJWt flyXsy kVbCJq hKmrdS dITRRO"
    >
      <div
        class="sc-bdnxRM sc-jSFjdj sc-hiKfDv flyXsy fkwVUy kvLUxc"
        direction="column"
      >
        <iframe
          class="sc-jJMGnK FDWTR"
          src="https://www.youtube.com/embed/nmXMgqjQzls"
        />
        <div
          class="sc-bdnxRM sc-jSFjdj sc-gXfVKN flyXsy fkwVUy kbtMsR"
          direction="column"
        >
          <h1
            class="sc-hKFxyN gcPkEz"
          >
            Video Tour | Welcome to Wizeline Guadalajara
          </h1>
          <p
            class="sc-eCApnc iaIVKm"
            font-size=".9rem"
          >
            Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...
          </p>
        </div>
      </div>
      <div
        class="sc-bdnxRM sc-gKAaRy sc-iklJeh flyXsy kVbCJq iceGUP"
      >
        <a
          class="sc-bYwzuL gbFdLf"
          href="/undefined"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s240-c-k-c0xffffffff-no-rj-mo"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Wizeline
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/nmXMgqjQzls"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/nmXMgqjQzls/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Video Tour | Welcome to Wizeline Guadalajara
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/HYyRZiwBWc8"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/HYyRZiwBWc8/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Wizeline Guadalajara | Bringing Silicon Valley to Mexico
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/Po3VwR_NNGk"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/Po3VwR_NNGk/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Wizeline hace sentir a empleados como en casa
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/7PtYNO6g7eI"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/7PtYNO6g7eI/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              We Are Wizeline
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/YuW0CE_8i1I"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/YuW0CE_8i1I/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Wizeline Thrives in Mexico City
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/CHzlSGRvWPs"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/CHzlSGRvWPs/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Wizeline
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/cjO2fJy8asM"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/cjO2fJy8asM/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              A Day in the Life of an Engineering Manager at Wizeline
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/zClI9OjgKXM"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/zClI9OjgKXM/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Wizeline Technical Writing Academy | Featuring Eduardo Ocejo
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/8bz9R61oY5o"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/8bz9R61oY5o/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Silicon Valley en México
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/7dJFraOqcoQ"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/7dJFraOqcoQ/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Why Wizeline? featuring Juan Pablo Villa in Mexico City
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/w-Qwc_XJrWc"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/w-Qwc_XJrWc/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Wizeline&#39;s 2020 Year in Review
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/rjir_cHTl5w"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/rjir_cHTl5w/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Why Wizeline? featuring Hugo Lopez in Mexico City
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/DcFK1x3NHGY"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/DcFK1x3NHGY/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Why Wizeline? (We&#39;re Hiring in Mexico!)
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/3BzYWAqZgFw"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/3BzYWAqZgFw/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Why Wizeline? featuring Oswaldo Herrera in Mexico City
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/3KVFmT-Tp2w"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/3KVFmT-Tp2w/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Caso de Éxito AWS: Wizeline [Spanish]
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/aKuPmY2m1Ro"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/aKuPmY2m1Ro/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Wizeline&#39;s 2019 Year in Review
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/24sTHUyWhRM"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/24sTHUyWhRM/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              [1 of 2] Wizeline CEO shares career lessons from Google
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/IxGc1gSqB3A"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/IxGc1gSqB3A/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Wizeline Data Engineering featuring Tania Reyes
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/NP1gAnbeNno"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/NP1gAnbeNno/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Wizeline Querétaro | Mexico&#39;s New Knowledge Economy (We&#39;re hiring!)
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/F6Krwu6lUc8"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/F6Krwu6lUc8/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Miriam Godinez | Women in Leadership at Wizeline
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/RFq7gfvhtCk"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/RFq7gfvhtCk/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Welcome Back to Wizeline Vietnam | Extended Version
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/E1Vq_A3WKK8"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/E1Vq_A3WKK8/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              How does Wizeline work?
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/ZmkslANDz0Q"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/ZmkslANDz0Q/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              12 Wishes from Wizeline | Happy Holidays 2019
            </p>
          </div>
        </a>
        <a
          class="sc-bYwzuL gbFdLf"
          href="/Nss3EmTDD3s"
        >
          <div
            class="sc-bdnxRM sc-jSFjdj sc-kLojOw flyXsy fkwVUy bRnREc"
            direction="column"
          >
            <img
              class="sc-fujyAs kyRtsq"
              src="https://i.ytimg.com/vi/Nss3EmTDD3s/mqdefault.jpg"
            />
            <p
              class="sc-eCApnc eGRAnL"
              font-size=".9rem"
              font-weight="bold"
            >
              Why Wizeline?
            </p>
          </div>
        </a>
      </div>
    </div>
  </main>
</div>
`;
